python/ray/util/sgd/torch/training_operator.py

```diff
@@ -344,18 +554,117 @@ def validate_batch(self, batch, batch_info):
 
     def state_dict(self):
         """Override this to return a representation of the operator state.
+        Any argument passed into self.register will automatically be saved.
+        Use this method to save any additional state.
 
         Returns:
             dict: The state dict of the operator."""
         pass
 
     def load_state_dict(self, state_dict):
         """Override this to load the representation of the operator state.
+        Anything passed into self.register will automatically be loaded. Use
+        this method to load any additional state.
 
         Args:
             state_dict (dict): State dict as returned by the operator. """
         pass
 
+    @staticmethod
+    def from_creators(model_creator, optimizer_creator, data_creator=None,
+                      scheduler_creator=None, loss_creator=None,
+                      serialize_data_creation=True):
```

Comment:
Need docstrings; should copy over from old diff

Reply:
done